core-tests/shared/src/test/scala/zio/stm/TPriorityQueueSpec.scala

```diff
@@ -15,6 +15,24 @@ object TPriorityQueueSpec extends ZIOBaseSpec {
         } yield values
         assertM(transaction.commit)(equalTo(vs.sorted))
       }
+    },
+    testM("removeIf") {
+      val elems = List(2, 4, 6, 3, 5, 6)
```

Comment:
you could possibly choose the even numbers out of order here to show evidence that the priority matters.

Reply:
Changed it to a property based test to cover that.